platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/SwirldState.java

```diff
@@ -65,24 +54,23 @@ default void init(
     }
 
     /**
-     * Given a round of consensus ordered events, update the state to reflect their effect. Events are iterated in
-     * consensus order by the {@link Round#iterator()}. Transactions in each event are iterated in consensus order by
-     * the {@link ConsensusEvent#consensusTransactionIterator()}. Transactions in a single event occur after all the
-     * transactions in the previous event and before all the transactions in the next event.
-     *
+     * Provides the application an opportunity to perform operations on transactions in an event prior to handling.
+     * Called against a given {@link Event} only once, globally (not once per state instance) This method may modify
+     * the {@link Transaction}s in the event by doing nothing, adding additional signatures, removing existing
+     * signatures, replacing signatures with versions that expand the public key from an application specific
+     * identifier to an actual public key, or attaching metadata. Additional signatures extracted from the transaction
+     * payload can also be added to the list of signatures to be verified.
      * <p>
-     * It is good if this method changes some class variables and then return. It is also OK if it spawns a number of
-     * threads that change those variables, then wait until all those threads have ended, and then return. It is even OK
-     * for it to create a pool of threads that continue to exist after it returns, as long as it ensures that those
-     * threads have finished all their changes before it returns. But it is an error for this method to spawn a thread
-     * that will make changes after the method returns and before the next time it is called. If this method creates
-     * threads that continue to exist after it returns (in a legal way), then it should clean up those resources when
-     * the object has been fully released (see documentation in {@link Releasable}). If the SwirldState extends one of
-     * the partial merkle implementations (e.g. PartialMerkleLeaf or PartialNaryMerkleInternal) then it can put that
-     * cleanup in the onDestroy() method, which is called when the object becomes fully released.
+     * <strong>This method is always invoked on an immutable state.</strong>
      *
-     * @param round           the round of consensus ordered events to update the state with
-     * @param swirldDualState current dualState object which can be read/written by the application
+     * @param event
+     * 		the event to perform pre-handling on
+     * @see #handleConsensusRound(Round, SwirldDualState)
+     */
+    default void preHandle(final Event event) {}
+
+    /**
+     * Handle transactions in a round apply them to the state.
      */
     void handleConsensusRound(final Round round, final SwirldDualState swirldDualState);
```

Comment:
I notice now that you deleted the parameter docs. Was that intentional?

Reply:
Added parameter javadocs